pa/fin/views.py

```diff
@@ -1,27 +1,54 @@
 """
 Views
 """
+import json
+import logging
+import traceback
 from decimal import Decimal, InvalidOperation
+from json import JSONDecodeError
 from threading import Thread
 
-from rest_framework import filters, viewsets
+from rest_framework import filters, viewsets, status
 from rest_framework.generics import get_object_or_404
 from rest_framework.metadata import SimpleMetadata
 from rest_framework.response import Response
 from rest_framework.views import APIView
 
-from .exceptions import BadRequest
-from .models.index import Index
-from .models.models import Account, Goal
+from .exceptions import BadRequest, TraderNetAPIUnavailable
+from .models.index import Index, Ticker
+from .models.models import Goal
+from .models.portfolio import Portfolio, PortfolioTickers, Account
 from .serializers.index import IndexSerializer, AdjustedTickerSerializer
-from .serializers.serializers import AccountSerializer, GoalSerializer
+from .serializers.portfolio import PortfolioSerializer, AccountSerializer
+from .serializers.serializers import GoalSerializer
 from .utils.index_helpers import update_tickers_industries
+from .utils.tradernet.PublicApiClient import PublicApiClient
+from .utils.tradernet.error_codes import BAD_SIGN
+
+logger = logging.getLogger(__name__)
 
 
 class AccountViewSet(viewsets.ModelViewSet):
     """
     API endpoint that allows accounts to be viewed or edited
     """
+
+    class AccountMetadata(SimpleMetadata):
+        def determine_metadata(self, request, view):
+            base_metadata = super().determine_metadata(request, view)
+            query = Portfolio.objects.filter(user=request.user).all()
+            base_metadata['actions']['POST']['portfolio'] = {
+                'type': 'choice',
+                'required': True,
+                'read_only': False,
+                'label': 'Portfolio',
+                'choices': [
```

Comment:
you can accomplish it using an own written serialization field that will give our choices as representation, it's more clear